tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -1650,9 +1650,10 @@ ParamModsFromIntrinsicArg(const HLSL_INTRINSIC_ARGUMENT *pArg) {
 }
 
 static void InitParamMods(const HLSL_INTRINSIC *pIntrinsic,
-                          SmallVectorImpl<hlsl::ParameterModifier> &paramMods) {
+                          SmallVectorImpl<hlsl::ParameterModifier> &paramMods,
+                          size_t uNumArgsWithVariadic = 0u) {
```

Comment:
nit: what about `uVariadicArgumentCount = 0`? 
As-is, it feels like we should set the number of arguments everytime, but default to 0, while it's only useful for variadic.

This of course requires to add a new size inside the function:
```
const size_t argumentCount = pIntrinsic->uNumArgs + uVariadicArgumentCount;
```

As for the loop, fine leaving it as you did, or if you have some time, maybe squash them into a single loop from 1 to argumentCount? But that's my personal preference to have a simple linear counter, not jumping around across multiple loops.

Reply:
I gave it a shot. The "numArgs" parameter now is closer to what it actually represents. We still have two loops though, but at least not with a shared index.

Reply:
Thanks!